src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1056,14 +1056,14 @@ protected internal struct EventData
             /// <param name="reserved">Value for reserved: 2 for per-provider metadata, 1 for per-event metadata</param>
             internal unsafe void SetMetadata(byte* pointer, int size, int reserved)
             {
-                this.m_Ptr = (long)(ulong)(UIntPtr)pointer;
+                this.m_Ptr = (ulong)pointer;
```

Comment:
I have tested this with a small app, byte* acts like an unsigned, and is zero extended.